packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -27,53 +28,41 @@ contract MixinWrapperFunctions is
     SafeMath
 {
   
-    /// @dev Fills an order with specified parameters and ECDSA signature. Throws if specified amount not filled entirely.
-    /// @param orderAddresses Array of order's maker, taker, makerToken, takerToken, and feeRecipient.
-    /// @param orderValues Array of order's makerTokenAmount, takerTokenAmount, makerFee, takerFee, expirationTimestampInSec, and salt.
-    /// @param takerTokenFillAmount Desired amount of takerToken to fill.
-    /// @param v ECDSA signature parameter v.
-    /// @param r ECDSA signature parameters r.
-    /// @param s ECDSA signature parameters s.
     function fillOrKillOrder(
         address[5] orderAddresses,
-        uint256[6] orderValues,
-        uint256 takerTokenFillAmount,
-        uint8 v,
-        bytes32 r,
-        bytes32 s)
+        uint[6] orderValues,
+        uint takerTokenFillAmount,
+        bytes signature)
         public
     {
         require(fillOrder(
             orderAddresses,
             orderValues,
             takerTokenFillAmount,
-            v,
-            r,
-            s
+            signature
         ) == takerTokenFillAmount);
     }
 
     /// @dev Fills an order with specified parameters and ECDSA signature. Returns false if the transaction would otherwise revert.
     /// @param orderAddresses Array of order's maker, taker, makerToken, takerToken, and feeRecipient.
     /// @param orderValues Array of order's makerTokenAmount, takerTokenAmount, makerFee, takerFee, expirationTimestampInSec, and salt.
     /// @param takerTokenFillAmount Desired amount of takerToken to fill.
-    /// @param v ECDSA signature parameter v.
-    /// @param r ECDSA signature parameters r.
-    /// @param s ECDSA signature parameters s.
     /// @return Success if the transaction did not revert.
     /// @return Total amount of takerToken filled in trade.
     function fillOrderNoThrow(
         address[5] orderAddresses,
         uint256[6] orderValues,
         uint256 takerTokenFillAmount,
-        uint8 v,
-        bytes32 r,
-        bytes32 s)
+        bytes signature)
         public
         returns (bool success, uint256 takerTokenFilledAmount)
     {
         bytes4 FILL_ORDER_FUNCTION_SIGNATURE = bytes4(keccak256("fillOrder(address[5],uint256[6],uint256,uint8,bytes32,bytes32)"));
         
+        // Input size is padded to a 4 + n * 32 byte boundary
```

Comment:
If this is not a WIP any more we need more comments. Much more comments.
I'm unable to read that. Let's reference to some solidity documentation or other sources that can help people understand that. (Talking about the whole function. Let's also have it as a rule to have a very good and verbose explanation before every assembly trick. How much does it save, why is it absolutely neccessary? not against assembly in general, but I want other people to be able to understand that even if they're not so experienced.

Reply:
Feel free to ignore it for now. It's still a prototype.

Reply:
Yeah, I'm not happy with the complexity of this assembly at all. It's not even tested and probably contains bugs. It does show that there is a solution though. We can reimplement it using `byte[] input` so we can construct the input arguments in Solidity instead of assembly.

Reply:
Added a `// TODO`s